dataans/src-tauri/src/dataans/command/import/json.rs

```diff
@@ -0,0 +1,69 @@
+use std::fs::File;
+use std::path::Path;
+use std::sync::Arc;
+
+use common::export::{Schema, SchemaV1};
+use futures::future::join_all;
+use serde_json;
+
+use crate::dataans::db::Db;
+use crate::dataans::service::note::NoteService;
+use crate::dataans::service::space::SpaceService;
+use crate::dataans::DataansError;
+
+pub async fn import_v1<D: Db>(
+    schema_v1: SchemaV1,
+    space_service: &Arc<SpaceService<D>>,
+    note_service: &Arc<NoteService<D>>,
+) -> Result<(), DataansError> {
+    let space_futures = schema_v1.data.into_iter().map(|space_data| {
+        let space_service = space_service.clone();
+        let note_service = note_service.clone();
+        async move {
+            let space = space_data.space;
+            let notes = space_data.notes;
+
+            if space_service.space_by_id(space.id).await.is_err() {
+                space_service.create_space(space).await?;
+            }
+
+            let note_futures = notes.into_iter().map(|note| {
+                let note_service = note_service.clone();
+                async move {
+                    if note_service.note_by_id(note.id).await.is_err() {
+                        note_service.create_note(note).await?;
+                    }
+                    Ok(())
+                }
+            });
+
+            join_all(note_futures)
+                .await
+                .into_iter()
+                .collect::<Result<Vec<_>, DataansError>>()?;
```

Comment:
I this code can be simplified. Can you try to use `try_join_all`? https://docs.rs/futures/latest/futures/future/fn.try_join_all.html

```suggestion
            try_join_all(note_futures)
                .await?;
```

As I remember, it should compile. Example: https://github.com/TheBestTvarynka/Dataans/blob/3dd790a16469e120ce5882dbb4c504e44b88f6cd/dataans/src-tauri/src/dataans/command/export/json.rs#L22-L29

Reply:
the same for `space_futures`